addons/docs/src/blocks/Source.tsx

```diff
@@ -32,13 +58,45 @@ type NoneProps = CommonProps;
 
 type SourceProps = SingleSourceProps | MultiSourceProps | CodeProps | NoneProps;
 
+const getSnippet = (
+  storyId: StoryId,
+  sourceContext: SourceContextProps,
+  docsContext: DocsContextProps
+): string => {
+  const { sources } = sourceContext;
+  const { storyStore } = docsContext;
+
+  const snippet = sources && sources[storyId];
+  const data = storyStore?.fromId(storyId);
+
+  if (data) {
+    const { parameters } = data;
+    // eslint-disable-next-line no-underscore-dangle
+    const isArgsStory = parameters.__isArgsStory;
+    const type = parameters.docs?.source?.type || SourceType.AUTO;
+
+    // if user has explicitly set this as a dynamic story, or this is an args story and there's a snippet
+    if (
+      type === SourceType.DYNAMIC ||
+      (type === SourceType.AUTO && snippet && isArgsStory && !parameters.docs?.source?.code)
```

Comment:
I'm not following this logic. Can you split it into parts?

Reply:
Updated the logic added comments to make it more readable. Better?

Reply:
Yes much better